src/bm_sim/switch.cpp

```diff
@@ -425,17 +425,27 @@ SwitchWContexts::swap_requested() {
   return false;
 }
 
+void
+SwitchWContexts::block_until_no_more_packets(cxt_id_t cxt_id) {
```

Comment:
You should stick with a function that does it for all contexts
- I cannot think of a use case where one would want to purge packets in a single context
- Because of the mutex, this function couldn't be used to implement  `block_until_no_more_packets_in_all_contexts`

Reply:
Yeah you are right. If we grab the mutex no new packets can be created in any context. Fixed.